include/dqlite/client.h

```diff
@@ -0,0 +1,361 @@
+ * instance's cluster.
+ *
+ * Each instance should call this function to describe the entire initial
+ * configuration, including itself. If this dqlite instance has previously
+ * contacted a cluster leader and refreshed its view of the cluster
+ * configuration, that information will supersede the description given using
+ * this function.
+ *
+ * @addr is the (abstract) address of the described node. @flags is the
+ * bitwise OR of zero or more of: #DQLITE_CLUSTER_ME, indicating that the
+ * declared address is the address of this node; #DQLITE_CLUSTER_BOOTSTRAP,
+ * indicating that the node with the declared address will be the first
+ * to run when the cluster starts up for the first time. Exactly one address
+ * should be declared with #DQLITE_CLUSTER_BOOTSTRAP, and the same boostrap
+ * address should be declared on all nodes.
+ *
+ * No reference to the string pointed to by @addr is kept after this function
+ * returns.
+ */
+int dqlite_cluster_declare_node(dqlite_cluster *cluster,
+				const char *addr,
+				int flags);
+
+/**
+ * Set the address on which this instance's server will listen for requests
+ * from client and messages from other servers.
+ *
+ * The format of the @addr argument must be one of
+ *
+ * 1. "<HOST>"
+ * 2. "<HOST>:<PORT>"
+ * 3. "@<PATH>"
+ *
+ * Where <HOST> is a numeric IPv4/IPv6 address, <PORT> is a port number, and
+ * <PATH> is an abstract Unix socket path. The port number defaults to 8080 if
+ * not specified. In the second form, if <HOST> is an IPv6 address, it must be
+ * enclosed in square brackets "[]". In the third form, if <PATH> is empty, the
+ * implementation will automatically select an available abstract Unix socket
+ * path, which can then be retrieved with dqlite_node_get_bind_address().
+ *
+ * If an abstract Unix socket is used the dqlite node will accept only
+ * connections originating from the same process.
+ *
+ * No reference to the memory pointed to by @address is kept, so any memory
+ * associated with them can be released after the function returns.
+ */
+int dqlite_cluster_set_server_bind_address(dqlite_cluster *cluster,
+					   const char *addr);
+
+/**
+ * Customize the function that this instance will use to connect to servers.
+ *
+ * The same callback will be used by this instance's client to contact servers,
+ * and by the instance's server to contact other servers. The callback may be
+ * invoked concurrently on multiple threads, and is responsible for handling
+ * any required synchronization internally.
+ */
+int dqlite_cluster_set_connect_func(dqlite_cluster *cluster,
+				    dqlite_connect_func connect,
+				    void *connect_arg);
+
+/**
+ * Configure the target number of voting nodes for this cluster.
+ *
+ * This must be an odd number.
+ */
+int dqlite_cluster_set_voters(dqlite_cluster *cluster, unsigned n);
+
+/**
+ * Configure the target number of standby nodes for this cluster.
+ */
+int dqlite_cluster_set_standbys(dqlite_cluster *cluster, unsigned n);
```

Comment:
What about all other knobs/functionality part of `dqlite_node`? `dqlite_node_enable_disk_mode()`, `dqlite_node_set_snapshot_params()`, `dqlite_node_set_network_latency()`, `dqlite_node_set_failure_domain()`, `dqlite_node_recover()` etc.

Will `dqlite_cluster` need to play catch-up with `dqlite_node` regarding knobs and functionality? (e.g. all the times we add a `dqlite_node_xxx` function we also need to add the equivalent `dqlite_cluster_xxx` function in order for dqlite C client users to "see" it).

At first sight it feels that `dqlite_cluster` is augmenting the API surface of `dqlite_node` instead of complementing it. That might be fine, not sure, but I first wanted to ask if it's intentional.